core/embed/firmware/bl_check.c

```diff
@@ -127,14 +149,24 @@ void check_and_replace_bootloader(void) {
 
   // replace bootloader with the latest one
   const uint32_t *data =
-      (const uint32_t *)&_binary_embed_firmware_bootloader_bin_start;
+      (const uint32_t
+           *)&_binary_embed_firmware_bootloaders_bootloader_zip_start;
   const uint32_t len =
-      (const uint32_t)&_binary_embed_firmware_bootloader_bin_size;
+      (const uint32_t)&_binary_embed_firmware_bootloaders_bootloader_zip_size;
+
+  struct uzlib_uncomp decomp = {0};
+  uint8_t decomp_window[UZLIB_WINDOW_SIZE] = {0};
+  uint8_t decomp_out[IMAGE_HEADER_SIZE] = {0};
```

Comment:
decomp_out should be aligned to a 32-bit boundary, as you later cast it to uint32_t*

Reply:
fixed https://github.com/trezor/trezor-firmware/pull/3304/commits/9143bf167745d860251791ae305470e98e153266